File: source/examples/CustomUserData/UpdateCustomUserData.m
RLMApp *app = [RLMApp appWithId:@"<YourAppId>"]; // replace this with your App ID
[app loginWithCredential:[RLMCredentials anonymousCredentials] completion:^(RLMUser *user, NSError *error) {
    if (error != nil) {
        NSLog(@"Failed to log in: %@", error);
        return;
    }
    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb-atlas"];
    RLMMongoDatabase *database = [client databaseWithName:@"my_database"];
    RLMMongoCollection *collection = [database collectionWithName:@"users"];

    // Update the user's custom data document
    [collection updateOneDocumentWhere:@{@"userId": [user identity]}
        updateDocument: @{@"favoriteColor": @"cerulean"}
        completion:^(RLMUpdateResult *updateResult, NSError *error) { 
            if (error != nil) {
                NSLog(@"Failed to insert: %@", error);
            }
            NSLog(@"Matched: %lu, modified: %lu", [updateResult matchedCount], [updateResult modifiedCount]); 
    }];
}];

File: source/examples/CustomUserData/UpdateCustomUserData.swift
let appId = "<YourAppId>" // replace this with your App ID
let app = App(id: appId)
app.login(credentials: Credentials.anonymous()) { (user, error) in
    guard error == nil else {
        print("Failed to log in: \(error!.localizedDescription)")
        return
    }
    guard let user = user else {
        fatalError("User is nil without error")
    }
    let client = user.mongoClient("mongodb-atlas")
    let database = client.database(named: "my_database")
    let collection = database.collection(withName: "users")
    collection.updateOneDocument(
        filter: ["userId": AnyBSON(user.identity!)],
        update: ["favoriteColor": "cerulean"]
    ) { (updateResult, error) in
          guard error == nil else {
              print("Failed to update: \(error!.localizedDescription)")
              return
          }
          print("Matched: \(updateResult!.matchedCount), updated: \(updateResult!.modifiedCount)")
    }
}